services/datamanager/builtin/sync/config.go

```diff
@@ -10,18 +10,81 @@ import (
 
 // Config is the sync config from builtin.
 type Config struct {
-	AdditionalSyncPaths        []string
-	CaptureDir                 string
-	CaptureDisabled            bool
+	// AdditionalSyncPaths defines the file system paths
+	// that should be synced in addition to the CaptureDir
+	// Generally 3rd party programs will write arbitrary
+	// files to these directories which are intended to be
+	// synced to the cloud by data manager.
+	AdditionalSyncPaths []string
+	// CaptureDir defines the file system path
+	// that data capture will write to and
+	// which data sync should sync from.
+	CaptureDir string
+	// CaptureDisabled, when true disables data capture.
+	// sync needs to know if capture is disabled b/c if it
+	// is not disabled sync will need to check if
+	// the disk has filled up enough that data sync needs to delete
+	// capture files without syncing them.
+	// See DeleteEveryNthWhenDiskFull for more info.
+	// defaults to filepath.Join(os.Getenv("HOME"), ".viam", "capture")
+	CaptureDisabled bool
+	// DeleteEveryNthWhenDiskFull defines the `n` in
+	// the psudocode:
+	// ```go
+	// captureEnabled := !config.CaptureDisabled
+	// if captureEnabled {
+	//     for {
+	//         time.Sleep(time.Second*30)
+	//         if diskFull() {
+	//             for i, file in range dataCaptureDirFiles {
+	//                 if fileIndex % n == 0 {
+	//                     delete file
+	//                 }
+	//             }
+	//         }
+	//     }
+	// }
+	// ```
+	//
+	// which in english reads:
+	//
+	// If datacapture is enabled then every 30 seconds
+	// if the disk is full (which is defined as the disk
+	// is 90% full and 50% is contributed by the CaptureDir)
+	// delete every Nth file in the CaptureDir & child
+	// directories.
+	//
+	// The intent is to prevent data capture from filling up the
+	// disk if the robot is unable to sync data for a long period
+	// of time. Defaults to 5.
 	DeleteEveryNthWhenDiskFull int
-	FileLastModifiedMillis     int
-	MaximumNumSyncThreads      int
-	ScheduledSyncDisabled      bool
-	SelectiveSyncerName        string
-	SyncIntervalMins           float64
-	Tags                       []string
+	// FileLastModifiedMillis defines the number of milliseconds that
+	// we should wait for an arbitrary file (aka a file that doesn't end in
+	// either the .prog nor the .capture file extension) before we consider
+	// it as being ready to sync.
+	// defaults to 10000, aka 10 seconds
+	FileLastModifiedMillis int
+	// MaximumNumSyncThreads defines the maximum number of goroutines which
+	// data sync should create to sync data to the cloud
+	// defaults to 1000
+	MaximumNumSyncThreads int
+	// ScheduledSyncDisabled, when true disables data capture syncing every SyncIntervalMins
```

Comment:
```suggestion
	// ScheduledSyncDisabled, when true disables data sync
```
?

Reply:
unfortunately not

If an SDK calls[ datamanager.Sync](https://github.com/viamrobotics/api/blob/8d4c37932c1f31d653b1202b18e5aa98835cf487/proto/viam/service/datamanager/v1/data_manager.proto#L15) then datamanager will sync.

Reply:
ah, good point. fine as-is, thanks